react/components/loaders/ShowMoreLoaderResult.js

```diff
@@ -19,16 +19,17 @@ const ShowMoreLoaderResult = props => {
   return (
     <SearchResult {...props}>
       <div className="vtex-search-result__button--show-more w-100 flex justify-center">
-        <Button
-          onClick={onFetchMore}
-          disabled={!(products.length < recordsFiltered)}
-          isLoading={fetchMoreLoading}
-          size="small"
-        >
-          <FormattedMessage
-            id="search-result.show-more-button"
-          />
-        </Button>
+        { (products.length < recordsFiltered) && (
```

Comment:
You should verify if the product is not undefined before call the property `length` because as you did it can throw an exception.